inc/container.class.php

```diff
@@ -1604,6 +1588,83 @@ public static function findContainer($itemtype, $type = 'tab', $subtype = '')
         return $id;
     }
 
+    public static function findContainers($itemtype, $type = 'tab', $subtype = '', $itemId = '')
+    {
+        /** @var DBmysql $DB */
+        global $DB;
+        $ids = [];
+
+        if (!empty($itemtype) && !empty($itemId) && class_exists($itemtype)) {
+
+            $obj = new $itemtype();
+            if ($obj->getFromDB($itemId)) {
+
+                $entityId = $obj->fields['entities_id'] ?? 0;
+                $entityIds = getAncestorsOf("glpi_entities", $entityId);
+                $entityIds[] = $entityId; // Add entity obj itself to the list
+                $glpiActiveEntities = $_SESSION['glpiactiveentities'] ?? 0;
+
+                $entityRestriction = getEntitiesRestrictCriteria('', '', $glpiActiveEntities, true, true);
+
+                if (count($entityIds) === 0) {
+                    $entityIds = [$entityId];
+                }
+                $entityIdList = implode(",", $entityIds);
+
+                $sql = "SELECT id FROM glpi_plugin_fields_containers
```

Comment:
You must use [GLPI DBIterator](https://glpi-developer-documentation.readthedocs.io/en/develop/devapi/database/dbiterator.html) instead of direct queries.